### DISCUSSION TITLE
Cant connect MPD server

### DISCUSSION THREAD

**Discussion by bibitocarlos** [jcorporation/myMPD#727] [Q&A] [answered]
myMPD version: [e.g. 8.1.0]

**Cant connect to mpd server with myMPD, works with Cantata.
I get this error "ERROR mympdapi MPD error: you don't have permission for"**

**Expected behavior**

To work ^^

**Server plattform (please complete the following information):**
 - OS: Archlinux
 - OS version Latest
 - Plattform:  x86_64
 - MPD version: Latest

**Client plattform (please complete the following information):**
 - Browser version Latest Firefox

**Debug logs (please attach if it can be usefull)**
`Apr 12 18:07:45 bibiserveur mympd[525803]: ERROR    mympdapi  MPD connection failed
Apr 12 18:07:45 bibiserveur mympd[525803]: ERROR    mympdapi  MPD error: you don't have permission for "idle" (9)
Apr 12 18:07:45 bibiserveur mympd[525803]: ERROR    mympdapi  MPD error: you don't have permission for "idle" (9)
Apr 12 18:07:45 bibiserveur mympd[525803]: ERROR    mympdapi  MPD error: you don't have permission for "idle" (9)
Apr 12 18:07:45 bibiserveur mympd[525803]: NOTICE   mpdworker Stopping mpd_worker thread
Apr 12 18:07:45 bibiserveur mympd[525803]: ERROR    mpdworker Cache update failed
Apr 12 18:07:45 bibiserveur mympd[525803]: ERROR    mpdworker MPD error: you don't have permission for "search" (9)
Apr 12 18:07:45 bibiserveur mympd[525803]: NOTICE   mpdworker MPD worker connected
Apr 12 18:07:45 bibiserveur mympd[525803]: NOTICE   mpdworker MPD worker connecting to socket "/run/mpd/socket"
Apr 12 18:07:45 bibiserveur mympd[525803]: NOTICE   mympdapi  Starting mpd_worker thread for INTERNAL_API_CACHES_CREATE
Apr 12 18:07:45 bibiserveur mympd[525803]: WARN     mympdapi  Disabling smart playlists feature
Apr 12 18:07:45 bibiserveur mympd[525803]: WARN     mympdapi  Disabling advanced queue feature, depends on mpd >= 0.24.0
Apr 12 18:07:45 bibiserveur mympd[525803]: NOTICE   mympdapi  Enabling position whence feature
Apr 12 18:07:45 bibiserveur mympd[525803]: NOTICE   mympdapi  Enabling delete playlist range feature
Apr 12 18:07:45 bibiserveur mympd[525803]: NOTICE   mympdapi  Enabling binarylimit feature
Apr 12 18:07:45 bibiserveur mympd[525803]: NOTICE   mympdapi  Enabling partitions feature
Apr 12 18:07:45 bibiserveur mympd[525803]: NOTICE   mympdapi  Enabling single oneshot feature
Apr 12 18:07:45 bibiserveur mympd[525803]: ERROR    mympdapi  MPD error: you don't have permission for "status" (9)
Apr 12 18:07:45 bibiserveur mympd[525803]: NOTICE   mympdapi  Enabled smartpls_generate_tag_list: Genre
Apr 12 18:07:45 bibiserveur mympd[525803]: NOTICE   mympdapi  Enabled tag_list_browse: Artist Album AlbumArtist Genre
Apr 12 18:07:45 bibiserveur mympd[525803]: NOTICE   mympdapi  Enabled tag_list_search: Artist Album AlbumArtist Title Genre
Apr 12 18:07:45 bibiserveur mympd[525803]: NOTICE   mympdapi  Enabled tag_list: Artist Album AlbumArtist Title Genre Disc Track Name
Apr 12 18:07:45 bibiserveur mympd[525803]: NOTICE   mympdapi  MPD supported tags: Artist Album AlbumArtist Title Track Name Genre Disc
Apr 12 18:07:45 bibiserveur mympd[525803]: WARN     mympdapi  Disabling library feature, music directory not defined
Apr 12 18:07:45 bibiserveur mympd[525803]: ERROR    mympdapi  Can't get music_directory value from mpd
Apr 12 18:07:45 bibiserveur mympd[525803]: ERROR    mympdapi  MPD error: you don't have permission for "config" (9)
Apr 12 18:07:45 bibiserveur mympd[525803]: NOTICE   mympdapi  MPD protocol version: 0.23.5
Apr 12 18:07:45 bibiserveur mympd[525803]: NOTICE   mympdapi  MPD connected
Apr 12 18:07:45 bibiserveur mympd[525803]: NOTICE   mympdapi  Connecting to MPD socket "/run/mpd/socket"
Apr 12 18:07:42 bibiserveur mympd[525803]: ERROR    mympdapi  MPD connection failed
Apr 12 18:07:42 bibiserveur mympd[525803]: ERROR    mympdapi  MPD error: you don't have permission for "idle" (9)
Apr 12 18:07:42 bibiserveur mympd[525803]: ERROR    mympdapi  MPD error: you don't have permission for "idle" (9)
Apr 12 18:07:42 bibiserveur mympd[525803]: ERROR    mympdapi  MPD error: you don't have permission for "idle" (9)
Apr 12 18:07:42 bibiserveur mympd[525803]: NOTICE   mpdworker Stopping mpd_worker thread
Apr 12 18:07:42 bibiserveur mympd[525803]: ERROR    mpdworker Cache update failed
Apr 12 18:07:42 bibiserveur mympd[525803]: ERROR    mpdworker MPD error: you don't have permission for "search" (9)
Apr 12 18:07:42 bibiserveur mympd[525803]: NOTICE   mpdworker MPD worker connected
Apr 12 18:07:42 bibiserveur mympd[525803]: NOTICE   mpdworker MPD worker connecting to socket "/run/mpd/socket"
Apr 12 18:07:42 bibiserveur mympd[525803]: NOTICE   mympdapi  Starting mpd_worker thread for INTERNAL_API_CACHES_CREATE
Apr 12 18:07:42 bibiserveur mympd[525803]: WARN     mympdapi  Disabling smart playlists feature
Apr 12 18:07:42 bibiserveur mympd[525803]: WARN     mympdapi  Disabling advanced queue feature, depends on mpd >= 0.24.0
Apr 12 18:07:42 bibiserveur mympd[525803]: NOTICE   mympdapi  Enabling position whence feature
Apr 12 18:07:42 bibiserveur mympd[525803]: NOTICE   mympdapi  Enabling delete playlist range feature
Apr 12 18:07:42 bibiserveur mympd[525803]: NOTICE   mympdapi  Enabling binarylimit feature
Apr 12 18:07:42 bibiserveur mympd[525803]: NOTICE   mympdapi  Enabling partitions feature
Apr 12 18:07:42 bibiserveur mympd[525803]: NOTICE   mympdapi  Enabling single oneshot feature
Apr 12 18:07:42 bibiserveur mympd[525803]: ERROR    mympdapi  MPD error: you don't have permission for "status" (9)
Apr 12 18:07:42 bibiserveur mympd[525803]: NOTICE   mympdapi  Enabled smartpls_generate_tag_list: Genre
Apr 12 18:07:42 bibiserveur mympd[525803]: NOTICE   mympdapi  Enabled tag_list_browse: Artist Album AlbumArtist Genre
Apr 12 18:07:42 bibiserveur mympd[525803]: NOTICE   mympdapi  Enabled tag_list_search: Artist Album AlbumArtist Title Genre
Apr 12 18:07:42 bibiserveur mympd[525803]: NOTICE   mympdapi  Enabled tag_list: Artist Album AlbumArtist Title Genre Disc Track Name
`

Look at https://jcorporation.github.io/myMPD/debug

 - myMPD debug log
 - MPD verbose log
 - Browser error console
 - Browser network console
 
**Configuration (please attach if it can be usefull)**

 `mpd.conf
` zeroconf_enabled        "yes"
mixer_type              "software"
volume_normalization    "yes"
filesystem_charset      "UTF-8"
audio_buffer_size       "8192"
log_file           "syslog"
pid_file "/var/lib/mpd/mpd.pid"
db_file "/var/lib/mpd/mpd.db"
state_file "/var/lib/mpd/mpdstate"
playlist_directory "/var/lib/mpd/playlists"
music_directory "/donnees/musique"
audio_output {
        type            "pipewire"
        name            "PipeWire Sound Server"
}
audio_output {
        type "pulse"
        name "pulse audio"
}
user            "mpd"
#group "users"
port            "6600"
bind_to_address          "192.168.1.11"
auto_update     "yes"
password "Big@read,add,control,admin"
connection_timeout     "120"
metadata_to_use         "AlbumArtist,Artist,Album,Title,Track,Disc,Genre,Name"
bind_to_address         "/run/mpd/socket"
neighbors {
        plugin          "udisks"
}
neighbors {
        plugin          "upnp"
}
# Ouput for http stream - myMPD local playback
audio_output {
        type            "httpd"
        name            "HTTP Stream"
        encoder         "lame" #to support safari on ios
        port            "8000"
        bitrate         "128"
        format          "44100:16:1"
        always_on   "yes"
        tags        "yes"
}
sticker_file            "/var/lib/mpd/sticker.sql"

**Comment by jcorporation** [comment, marked as answer]
You must set the mpd password in the connection settings.
